src/containers/BooksForm.js

```diff
@@ -10,31 +14,76 @@ const category = [
   { id: 7, name: 'Sci-fi' },
 ];
 
-const BooksForm = () => (
-  <div>
-    <form>
-      <label htmlFor="title">
-        The Title of the Book
-        <input
-          type="text"
-          placeholder="enter the title of the book"
-          name="title"
-          id="title"
-        />
-      </label>
-      <label htmlFor="select">
-        <select>
-          {category.map(c => (
-            <option key={c.id}>{c.name}</option>
-          ))}
-        </select>
-      </label>
-      <button type="submit" value="submit">
-        Submit
-      </button>
-    </form>
-  </div>
-);
-
-
-export default BooksForm;
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange = (ev) => {
+    ev.preventDefault();
+    const { value, name } = ev.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit = (ev) => {
+    ev.preventDefault();
+    const { title, category } = this.state;
+    const id = Math.floor(Math.random() * 100);
+    const { createBook } = this.props;
+    createBook({ id, title, category });
+
+    this.setState({
+      title: '',
+      category: '',
+    });
+  }
+
+  render() {
+    const { title } = this.state;
+    return (
+      <div>
+        <form>
+          <label htmlFor="title">
+            The Title of the Book
+            <input
+              type="text"
+              placeholder="enter the title of the book"
+              value={title}
+              name="title"
+              id="title"
+              onChange={this.handleChange}
+            />
+          </label>
+          <label htmlFor="select">
+            <select name="category" onChange={this.handleChange}>
+              {category.map(c => (
+                <option key={c.id}>{c.name}</option>
+              ))}
+            </select>
+          </label>
+          <button type="submit" onClick={this.handleSubmit} value="submit">
+            Submit
+          </button>
+        </form>
```

Comment:
[Optional]

I think it would be better to use the default HTML5 form validation here. It's would prevent submitting empty data as shown below.

My suggestion would be to move the `onClick={this.handleSubmit}` inside the form element as `<form onSubmit={this.handleSubmit}>`, this way on `submit`, the form validation will check for invalid `inputs` before submitting the data. You would, however, need to add the `required` attribute to the input elements, bonus points, you can add other validations like the input maximum and minimum length, including the format you would want.



| Current behaviour | Suggested Behaviour |
| :-----: | :-----: |
| ![record1](https://user-images.githubusercontent.com/36057474/83012835-e2e98900-a013-11ea-8e0f-10a6deff909e.gif)  |  ![record1](https://user-images.githubusercontent.com/36057474/83013711-5cce4200-a015-11ea-9ad7-0a8ccc959347.gif) |